File: Algorithm_graduate/Java_OFFICER/Flow.java
import java.util.ArrayList;

public class Flow {
	final static float MAX_WEIGHT = 10;
	final static float MAX_PF = 5; // 5Mbps
	int seq = 0;
	float weight = 0;
	int innode = 0, outnode = 0;
	float pf = 0;
	ArrayList<Integer> DefaultPath_Switches = new ArrayList<Integer>();
	ArrayList<Integer> DefaultPath_Links = new ArrayList<Integer>();
	public void initiate(int seq,int node_number) {
		this.seq = seq;
		weight = (float)(Math.random() * MAX_WEIGHT); //Randomize weight to 0.0~10.0
		pf = (float)(Math.random() * MAX_PF); //Randomize pf to 0.0~5.0Mbps
		while (innode == outnode) {
			innode = (int)(Math.random() * node_number + 1);
			outnode = (int)(Math.random() * node_number + 1);
		}
	}
}

File: Python_Test/com/ruleplacement/Basic/flow.py
import random

class Flow:
    count = 0
    max_weight = 10 # Just a metric for the importance of a flow
    max_pf = 5 # 5Mbps

    def __init__(self, switch_number = 8):
        Flow.count = Flow.count + 1
        self.seq = Flow.count
        self.weight = 0
        self.pf = 0
        self.DefaultPath_links = None
        self.DefaultPath_switches = None
        self.innode, self.outnode = self.getRandomInOut(switch_number)

    def getRandomWeight(self, seed = 10):
        random.seed(seed)
        return Flow.max_weight * random.random()

    def getRandomInOut(self,sw_number = 8):
        self.innode, self.outnode = 0, 0
        random.seed(20 * self.seq)
        while self.innode == self.outnode:
            self.innode = int(sw_number * random.random() + 1)
            self.outnode = int(sw_number * random.random() + 1)
        return self.innode, self.outnode

    def showInfo(self):
        print 'Flow %d: weight %f, packet rate %fMbps, in-out set (%d, %d)' \
              %(self.seq, self.weight, self.pf, self.innode, self.outnode)